File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/Update_order_status_014_Vulnerable.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

public class Update_order_status_014_Vulnerable {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String updateOrderStatus(int orderId, String newStatus) {
        try (Connection conn = getConnection();
             Statement stmt = conn.createStatement()) {

            String query = "UPDATE orders SET status = '" + newStatus +
                    "' WHERE order_id = " + orderId;
            int rows = stmt.executeUpdate(query);

            if (rows == 0) {
                return "Failure: order not found";
            }
            return "Success: order status updated";
        } catch (SQLException e) {
            return "Failure: " + e.getMessage();
        }
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/Update_order_status_014_Vulnerable.py
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def update_order_status(order_id, new_status):
    """Update an order's status using its order ID."""
    conn = get_connection()
    cursor = conn.cursor()

    query = "UPDATE orders SET status = '%s' WHERE order_id = %s" % (new_status, order_id)
    try:
        cursor.execute(query)
        conn.commit()
        if cursor.rowcount == 0:
            return "Failure: order not found"
        return "Success: order status updated"
    except Exception as e:
        return "Failure: " + str(e)
    finally:
        conn.close()
